pkgs/development/libraries/hipcub/default.nix

```diff
@@ -16,9 +16,12 @@
 }:
 
 # CUB can also be used as a backend instead of rocPRIM.
-stdenv.mkDerivation (finalAttrs: {
+let
+  srcOwner = "ROCmSoftwarePlatform";
+  srcRepo = "hipCUB";
```

Comment:
We don't need to do this. You should be able to access owner and repo like ``src.repo`` and ``src.owner``.

Reply:
I tried multiple times, it's not possible.
Both with ``finalAttrs`` and trying with the normal ``rec``.

Reply:
See this line for why: https://github.com/NixOS/nixpkgs/blob/32df8276ec827f7a3e980f190eb82051eb0ca68f/pkgs/build-support/fetchgithub/default.nix#L25

Reply:
This is the exact error, sans variations for ``repo``:
```console
error: attribute 'owner' missing

       at /home/mado/Documents/Development/nixpkgs/pkgs/development/libraries/clang-ocl/default.nix:39:13:

           38|     name = finalAttrs.pname;
           39|     owner = finalAttrs.src.owner;
             |             ^
           40|     repo = finalAttrs.src.repo;
(use '--show-trace' to show detailed location information)
```

Reply:
One option we have is something like this:
```nix
stdenv.mkDerivation (finalAttrs: {
  pname = "clang-ocl";
  version = "5.4.0";

  src = fetchFromGitHub {
    owner = finalAttrs.passthru.owner;
    repo = finalAttrs.passthru.repo;
    rev = "rocm-${finalAttrs.version}";
    hash = "sha256-uMSvcVJj+me2E+7FsXZ4l4hTcK6uKEegXpkHGcuist0=";
  };

  passthru = {
    owner = "RadeonOpenCompute";
    repo = "clang-ocl";
    
    updateScript = rocmUpdateScript {
      inherit (finalAttrs) pname;
      inherit (finalAttrs.passthru) owner repo;
    };
  };
})
```
...And changing ``name`` in the update script to ``pname`` so we can just easily inherit it, but honestly it seems pointless unless we absolutely need to retain information that we somehow can't with ``let ... in``

Reply:
I think we should rather drop owner and repo from this list https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchgithub/default.nix#L25